arroyo-df/src/physical.rs

```diff
@@ -156,7 +154,119 @@ pub fn window_scalar_function() -> ScalarUDF {
     )
 }
 
-impl FunctionRegistry for EmptyRegistry {
+#[derive(WrapperApi)]
+struct UdfDylibInterface {
+    run: unsafe extern "C" fn(
+        args: Vec<(FFI_ArrowArray, FFI_ArrowSchema)>,
+    ) -> (FFI_ArrowArray, FFI_ArrowSchema),
+}
+
+pub struct UdfDylib {
+    name: String,
+    signature: Signature,
+    return_type: DataType,
+    udf: Container<UdfDylibInterface>,
+}
+
+impl Debug for UdfDylib {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("UdfDylib").finish()
+    }
+}
+
+impl UdfDylib {
+    pub async fn new(name: &str, config: &DylibUdfConfig) -> Self {
```

Comment:
I wouldn't expect something called "new" to go and get data from a remote environment and write it to the file system. possibly "download" or something?

Reply:
Renamed to `init` and added a comment.